### PR TITLE
Testcase for recent PR #1656

### PR DESCRIPTION
**Pull request by b-scholz**
The code generation for indices with different attribute types failed; this test-case checks for this scenario.

### CODE REVIEW

File: tests/semantic.at [hunk 1]
```diff
@@ -112,6 +112,7 @@ NEGATIVE_TEST([comp_types],[semantic])
 POSITIVE_TEST([comp_opt],[semantic])
 NEGATIVE_TEST([counter2],[semantic])
 POSITIVE_TEST([counter],[semantic])
+POSITIVE_TEST([cmp],[semantic])
```
Comment (mmcgr) [inline tests/semantic.at:115]: A more descriptive name would be great - `comparator`, or even 'comparator_indirect`.

File: tests/semantic/cmp/cmp.dl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Souffle - A Datalog Compiler
+// Copyright (c) 2020, The Souffle Developers. All rights reserved
+// Licensed under the Universal Permissive License v 1.0 as shown at:
+// - https://opensource.org/licenses/UPL
+// - <souffle root>/licenses/SOUFFLE-UPL.txt
+
+// Index comparator test for attributes of different type
```
Comment (mmcgr) [inline tests/semantic/cmp/cmp.dl:7]: The comment could be expanded - perhaps explaining that the test is for indirect index generation which is expected to change at 6+ attributes.

 Currently this tests indirect indices, but a change in the code would leave this testing direct indices and there would be no indication that this test was no longer fulfilling its purpose.

**Review by mmcgr — COMMENTED** [review]
I think this would be better in evaluation/fastevaluation, but if you're sure about semantic, that's fine too.

Would it be worth extending the test to other arities?

**Review by mmcgr — CHANGES_REQUESTED** [review]
Slightly more information in the name and comment would really help this test to keep being useful.

File: tests/semantic/cmp/cmp.dl (hunk 2 above)
Comment (b-scholz) [reply tests/semantic/cmp/cmp.dl]: I renamed the files and added the comment.
Unfortunately, constructing a test case with more attributes is quite involved. 
We would need to construct an example for which the RamDomain (int32_t) comparator states exactly the opposite of the attribute comparator.

File: tests/fastevaluation.at [hunk 3]
```diff
@@ -16,6 +16,7 @@ POSITIVE_TEST([cat],[evaluation])
 POSITIVE_TEST([comp-override1],[evaluation])
 POSITIVE_TEST([comp-override2],[evaluation])
 POSITIVE_TEST([comp-override3],[evaluation])
+POSITIVE_TEST([comperator_indirect],[evaluation])
```
Comment (mmcgr) [inline tests/fastevaluation.at:19]:
```suggestion
POSITIVE_TEST([comparator_indirect],[evaluation])
```

**Review by mmcgr — CHANGES_REQUESTED** [review]
Looks good, just one typo fix to go.

**Review by mmcgr — APPROVED** [review]

**Pull request merged**